### PR TITLE
Change copy-paste function to handle string constants (follow up PR)

### PR DESCRIPTION
**Pull request by Siedlerchr**
We changed the copy and paste function to handle string constants automatically.

    When copying a .bib reference, the string constants are added to the clipboard.
    When pasting a .bib reference with string constants, these are processed and added to the library database.
        If there are duplicate string constants, the user will be notified (error dialog) and the older string constants will be kept.

This was a quality of life features requested from a user and improves the workflow of the application. However, the user suggested originally to add a separate copy-string-constants option to the UI. We opted instead to add this feature automatically when copying with string constants to make the action more seamless.

Warnings on copy paste with duplicates
grafik

Closes https://github.com/JabRef/jabref/issues/10872

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
TODO: Clipbaord tests
org.jabref.gui.ClipBoardManagerTest

  Test Check that the ClipBoardManager can set a bibentry as its content from the clipboard FAILED

  java.lang.NullPointerException: Cannot invoke "java.util.SequencedSet.stream()" because the return value of "org.jabref.model.entry.BibEntryType.getRequiredFields()" is null

  Test Check that the ClipBoardManager can handle a bibentry with string constants correctly from the clipboard FAILED

  java.lang.NullPointerException: Cannot invoke "java.util.SequencedSet.stream()" because the return value of "org.jabref.model.entry.BibEntryType.getRequiredFields()" is null

File: CHANGELOG.md [hunk 1]
```diff
@@ -47,6 +48,7 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 - We enhanced the dialog for adding new fields in the content selector with a selection box containing a list of standard fields. [#10912](https://github.com/JabRef/jabref/pull/10912)
 - We store the citation relations in an LRU cache to avoid bloating the memory and out-of-memory exceptions. [#10958](https://github.com/JabRef/jabref/issues/10958)
 - Keywords filed are now displayed as tags. [#10910](https://github.com/JabRef/jabref/pull/10910)
+- When pasting, string constants are automatically added to the library database, while referenced constants are added to the clipboard during copying. [#10872](https://github.com/JabRef/jabref/issues/10872)
```
Comment (calixtus) [inline CHANGELOG.md:51]: Why two entries in the changelog?

Comment (Siedlerchr) [reply CHANGELOG.md]: One for changed behavior and one for new feature

**Review by koppor — APPROVED** [review]

**Review by calixtus — APPROVED** [review]
LGTM
Follow up would be adding the shift key as modifier to paste without constants. 😝

**Comment by Siedlerchr** [comment]
> We opted instead to add this feature automatically when copying with string constants to make the action more seamless.

**Pull request merged**